extension/nine-patch/src/commonMain/kotlin/com/seiko/imageloader/util/NinePatchPainter.kt

```diff
@@ -0,0 +1,221 @@
+            colorFilter = colorFilter,
+            filterQuality = filterQuality,
+        )
+        // Bottom
+        drawImage(
+            image,
+            srcOffset = offsetBottom,
+            srcSize = sizeBottom,
+            dstOffset = IntOffset(scaleLeft, drawHeight - scaleBottom),
+            dstSize = IntSize(scaleWidth, scaleBottom),
+            alpha = alpha,
+            colorFilter = colorFilter,
+            filterQuality = filterQuality,
+        )
+        // TopLeft
+        drawImage(
+            image,
+            srcOffset = offsetTopLeft,
+            srcSize = sizeTopLeft,
+            dstOffset = IntOffset.Zero,
+            dstSize = IntSize(scaleLeft, scaleTop),
+            alpha = alpha,
+            colorFilter = colorFilter,
+            filterQuality = filterQuality,
+        )
+        // TopRight
+        drawImage(
+            image,
+            srcOffset = offsetTopRight,
+            sizeTopRight,
+            dstOffset = IntOffset(drawWidth - scaleRight, 0),
+            dstSize = IntSize(scaleRight, scaleTop),
+            alpha = alpha,
+            colorFilter = colorFilter,
+            filterQuality = filterQuality,
+        )
+        // BottomLeft
+        drawImage(
+            image,
+            srcOffset = offsetBottomLeft,
+            srcSize = sizeBottomLeft,
+            dstOffset = IntOffset(0, drawHeight - scaleBottom),
+            dstSize = IntSize(scaleLeft, scaleBottom),
+            alpha = alpha,
+            colorFilter = colorFilter,
+            filterQuality = filterQuality,
+        )
+        // BottomRight
+        drawImage(
+            image,
+            srcOffset = offsetBottomRight,
+            srcSize = sizeBottomRight,
+            dstOffset = IntOffset(drawWidth - scaleRight, drawHeight - scaleBottom),
+            dstSize = IntSize(scaleRight, scaleBottom),
+            alpha = alpha,
+            colorFilter = colorFilter,
+            filterQuality = filterQuality,
+        )
+    }
+
+    override val intrinsicSize: Size
+        get() = Size.Unspecified
+
+    override fun applyAlpha(alpha: Float): Boolean {
+        this.alpha = alpha
+        return true
+    }
+
+    override fun applyColorFilter(colorFilter: ColorFilter?): Boolean {
+        this.colorFilter = colorFilter
+        return true
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is NinePatchPainter) return false
+        if (image != other.image) return false
+        if (ninePatchData != other.ninePatchData) return false
+        if (scale != other.scale) return false
+        if (filterQuality != other.filterQuality) return false
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = image.hashCode()
+        result = 31 * result + ninePatchData.hashCode()
+        result = 31 * result + scale.hashCode()
+        result = 31 * result + filterQuality.hashCode()
```

Comment:
`scale` and `filterQuality` are part of `ninePatchData` and rather do not need separate participation in the hash function